### PR TITLE
`prefer-string-slice`: Keep optional chaining in autofix

### LINKED ISSUE #999: Broken autofix in prefer-string-slice with optional chaining
When fixing code with the `prefer-string-slice` rule, code which contains the optional chaining operator can break, as the operator is discarded in the output.

This occurs when using the recommended rules.

**Example input and configuration**
```js
const foo = {};
console.log(foo.bar?.substring(2));
```
```json eslintrc
{
	"plugins": [
		"unicorn"
	],
	"parserOptions": {
		"ecmaVersion": 2020
	},
	"rules": {
		"unicorn/prefer-string-slice": "error"
	}
}
```

Expected output after the auto fix runs is `foo.bar?.slice(2)`, and the program should output `undefined`.

Observed output does not retain optional chaining operator, and after fixing is `foo.bar.slice(2)`. This causes the above program to throw an error on execution instead of outputting `undefined`.

Resources:
* [Optional chaining operator on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)
* [OptionalExpression Specification](https://tc39.es/ecma262/#prod-OptionalExpression)
* [Feature support](https://node.green/#ES2020-features-optional-chaining-operator-----) (tldr: all current Node versions)

### PR DESCRIPTION
**Pull request by noftaly**
Keep optional chaining in auto fix

Fixes #999

### CODE REVIEW

**Comment by fisker** [comment]
What about  `foo?.slice?.(1, 2)`

**Comment by fisker** [comment]
And  `foo?.bar.slice(1, 2)`, I believe it's different `callee` type.

**Comment by noftaly** [comment]
> And  `foo?.bar.slice(1, 2)`, I believe it's different `callee` type.

This is already tested

> What about foo?.slice?.(1, 2)

I'll add this one in an hour or so

**Review by sindresorhus — APPROVED** [review]

File: test/prefer-string-slice.js [hunk 1]
```diff
@@ -32,6 +35,46 @@ test({
...
+			errors: errorsSubstring
+		},
+		{
+			code: 'foo?.[0]?.substring()',
+			output: 'foo?.[0]?.slice()',
+			errors: errorsSubstring
+		},
+		{
+			code: 'foo.bar.substr?.()',
+			output: 'foo.bar.slice?.()',
```
Comment (fisker) [inline test/prefer-string-slice.js:55]: I'm not sure about this, the old one is testing object has `.substr`, but new one testing `.slice` exists, if someone mean to test it is a `string` and `foo.bar` is an `array`, it's broken.

Comment (fisker) [reply test/prefer-string-slice.js:55]: Hope no one use this...

**Review by fisker — APPROVED** [review]

File: test/prefer-string-slice.js (hunk 1 above)
Comment (noftaly) [reply test/prefer-string-slice.js:55]: As you want

**Pull request merged**